contracts/extensions/VotingReputationFactory.sol

```diff
@@ -0,0 +1,48 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "./../ColonyDataTypes.sol";
+import "./../IColony.sol";
+import "./ExtensionFactory.sol";
+import "./VotingReputation.sol";
+
+
+contract VotingReputationFactory is ExtensionFactory, ColonyDataTypes { // ignore-swc-123
+  mapping (address => VotingReputation) public deployedExtensions;
+
+  modifier isRoot(address _colony) {
+    require(IColony(_colony).hasUserRole(msg.sender, 1, ColonyRole.Root), "colony-extension-user-not-root"); // ignore-swc-123
+    _;
+  }
+
+  function deployExtension(address _colony) external isRoot(_colony) {
+    require(deployedExtensions[_colony] == VotingReputation(0x00), "colony-extension-already-deployed");
+    VotingReputation newExtensionAddress = new VotingReputation(_colony);
```

Comment:
Might be worth thinking about establishing an EtherRouter pattern here. I know we didn't for the other extension contracts, but they were much smaller.

Reply:
True, but in my mind since extensions can be attached/detached, the "upgrade" process for extensions is to detach the old one and attach the new one -- there's no indefinite state we need to protect, unlike for the Colony itself. So we may be able to avoid the EtherRouter complexity/gas overhead here. I'm open to argument though.

Reply:
It was purely from a deployment cost perspective I was thinking about this, but I guess given we expect it to sit around forever, the extra per-tx cost of using an EtherRouter would eventually outweigh the benefits gained.

Reply:
Remind me how the deployment cost is lower? For the first deployment or amortized over many upgrades?

Also, now that I think about it, including a `suicide onlyOwner { suicide(owner) }` function which can be called by the colony seems like not a terrible idea -- I can imagine when uninstalling an extension we should also be able to free up  the storage.

Reply:
Deploying an EtherRouter that points at an already-set-up Resolver is much cheaper than deploying a new instance of `VotingReputation`. I will admit to having my colonyNetwork blinkers on here though - I think the correct way to reduce the deployment cost here is via libraries.

Reply:
How would you feel about leaving this optimization to a future PR? Seems like a larger review of extension patters (i.e. installing to domains, destroying on uninstall, library/etherrouter) would be useful soonish, but perhaps separate from the voting logic itself?